### PR TITLE
Read the Generators from portal.dat file 0x0E00000D and enhance DatRe…

### PR DESCRIPTION
**Pull request by Lidefeath**
…ader to handle obfuscated strings.

### CODE REVIEW

File: Source/ACE.DatLoader/DatReader.cs [hunk 1]
```diff
@@ -93,16 +93,37 @@ public float ReadSingle()
...
             byte[] thestring = new byte[stringlength];
             Array.Copy(Buffer, Offset, thestring, 0, stringlength);
             Offset += stringlength;
             return System.Text.Encoding.ASCII.GetString(thestring);
         }
 
+        /// <summary>
+        /// Returns a string as defined by the first byte's length and removes the obfuscation
+        /// </summary>
+        public string ReadOString()
```
Comment (Mogwai-AC) [inline Source/ACE.DatLoader/DatReader.cs]: What's an "OString"?  heck, what's a "PString" for that matter.  I realize you're deobfuscating strings, but let's deobfuscate method names while we're at it.

**Comment by Lidefeath** [comment]
Ok, changes are in.

File: Source/ACE.DatLoader/DatReader.cs [hunk 2]
```diff
@@ -93,16 +93,37 @@ public float ReadSingle()
...
             byte[] thestring = new byte[stringlength];
             Array.Copy(Buffer, Offset, thestring, 0, stringlength);
             Offset += stringlength;
             return System.Text.Encoding.ASCII.GetString(thestring);
         }
 
+        /// <summary>
+        /// Returns a string as defined by the first byte's length and removes the obfuscation
+        /// </summary>
+        public string ReadObfuscatedString()
```
Comment (Zegeger) [inline Source/ACE.DatLoader/DatReader.cs]: I think a better name would be ReadBigEndianString

File: Source/ACE.DatLoader/DatReader.cs [hunk 3]
```diff
@@ -93,16 +93,37 @@ public float ReadSingle()
             return data;
         }
 
+        /// <summary>
+        /// Returns a string as defined by the first byte's length
+        /// </summary>
         public string ReadPString()
```
Comment (Zegeger) [inline Source/ACE.DatLoader/DatReader.cs]: PString just seems to be their version of a basic string.  This seems to be identical to ReadString except it reads the length from the first byte.  I don't see any references to ReadString being used, so should this just replace ReadString?

**Pull request merged**